prow/cmd/pipeline/dev.yaml

```diff
@@ -0,0 +1,119 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: prow-pipeline
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prow-pipeline
+  template:
+    metadata:
+      labels:
+        app: prow-pipeline
+    spec:
+      serviceAccount: prow-pipeline
+      containers:
+      - name: pipeline
+        image: gcr.io/k8s-testimages/pipeline:latest  # Note: not gcr.io/k8s-prow for dev
+        imagePullPolicy: Always  # Good practice for dev/debugging, bad for prod
```

Comment:
This is great for prod when using a floating tag like `latest` if you want updates to percolate and ensure node image caches are not making prod deployments change based on target node

Reply:
This deployment is intended only to be used during development. That's why is using `Always` with `latest`. 

I will follow up with a production deployment in a different PR with will use a tag.